presto-oracle/src/test/java/io/prestosql/plugin/oracle/BaseOracleIntegrationSmokeTest.java

```diff
@@ -113,39 +157,65 @@ public void testPredicatePushdown()
-        predicatePushdownTest("decimal(9, 3)", "123.321", "<=", "123.321", "CAST(123.321 AS decimal(9, 3))");
-        predicatePushdownTest("decimal(9, 3)", "123.321", "=", "123.321", "CAST(123.321 AS decimal(9, 3))");
-        predicatePushdownTest("decimal(30, 10)", "123456789.987654321", "<=", "123456790", "CAST(123456789.987654321 AS decimal(30, 10))");
-        predicatePushdownTest("decimal(30, 10)", "123456789.987654321", "<=", "123456789.987654321", "CAST(123456789.987654321 AS decimal(30, 10))");
-        predicatePushdownTest("decimal(30, 10)", "123456789.987654321", "=", "123456789.987654321", "CAST(123456789.987654321 AS decimal(30, 10))");
-        predicatePushdownTest("float(63)", "123456789.987654321", "<=", "CAST(123456789.99 AS REAL)", "CAST(123456789.987654321 AS DOUBLE)");
-        predicatePushdownTest("float(63)", "123456789.987654321", "<=", "CAST(123456789.99 AS DOUBLE)", "CAST(123456789.987654321 AS DOUBLE)");
-        predicatePushdownTest("float(126)", "123456789.987654321", "<=", "CAST(123456789.99 AS REAL)", "CAST(123456789.987654321 AS DOUBLE)");
-        predicatePushdownTest("float(126)", "123456789.987654321", "<=", "CAST(123456789.99 AS DOUBLE)", "CAST(123456789.987654321 AS DOUBLE)");
-        predicatePushdownTest("CHAR(1)", "'0'", "=", "'0'", "CHAR'0'");
-        predicatePushdownTest("CHAR(1)", "'0'", "<=", "'0'", "CHAR'0'");
-        predicatePushdownTest("CHAR(5)", "'0'", "=", "CHAR'0'", "CHAR'0    '");
+    @Test
+    public void testPredicatePushdownForNumerics()
+    {
+        predicatePushdownTest("decimal(9, 3)", "123.321", "<=", "124");
+        predicatePushdownTest("decimal(9, 3)", "123.321", "<=", "123.321");
+        predicatePushdownTest("decimal(9, 3)", "123.321", "=", "123.321");
+        predicatePushdownTest("decimal(30, 10)", "123456789.987654321", "<=", "123456790");
+        predicatePushdownTest("decimal(30, 10)", "123456789.987654321", "<=", "123456789.987654321");
+        predicatePushdownTest("decimal(30, 10)", "123456789.987654321", "=", "123456789.987654321");
+        predicatePushdownTest("float(63)", "123456789.987654321", "<=", "CAST(123456789.99 AS REAL)");
+        predicatePushdownTest("float(63)", "123456789.987654321", "<=", "CAST(123456789.99 AS DOUBLE)");
+        predicatePushdownTest("float(126)", "123456789.987654321", "<=", "CAST(123456789.99 AS REAL)");
+        predicatePushdownTest("float(126)", "123456789.987654321", "<=", "CAST(123456789.99 AS DOUBLE)");
+        predicatePushdownTest("BINARY_FLOAT", "5.0f", "=", "CAST(5.0 AS REAL)");
+        predicatePushdownTest("BINARY_DOUBLE", "20.233", "=", "CAST(20.233 AS DOUBLE)");
+        predicatePushdownTest("NUMBER(5,3)", "5.0", "=", "CAST(5.0 AS DECIMAL(5,3))");
     }
 
-    private void predicatePushdownTest(String oracleType, String oracleLiteral, String operator, String filterLiteral, String compareLiteral)
-            throws Exception
+    @Test
+    public void testPredicatePushdownForChars()
```

Comment:
is it just method split?

wonder whether this is covered by `io.prestosql.testing.AbstractTestDistributedQueries#testDataMappingSmokeTest`

(perhaps not full?)

Reply:
There is some overlap, but this method additionally checks that predicates are pushed dow in the presence of casts.

Reply:
oh, i missed that -- the method tests that the PPD _actually_ takes place, not only that results are correct